### PR TITLE
#1976 Document Builder. Lists

### PR DESCRIPTION
**Pull request by BALEHOK**
Relates to #1976 

- [x] Allow Row and Col as list elements
- [x] Fix unit tests
- [x] Code cleanup

Edit page:
image

Side panel:
image

### CODE REVIEW

**Comment by twschiller** [comment]
Testing notes

We probably don't need to support this out of the gate, but I noticed manually providing a list errors out:

image

**Comment by BALEHOK** [comment]
Manually created array as a source for the list
image

If a list fails to render the rest of the panel is still there
image

File: src/components/documentBuilder/DocumentList.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        })
+      ),
+    [array, elementKey, config, documentContext]
+  );
+
+  return isLoading ? (
+    <GridLoader />
+  ) : (
+    <>
+      {rootDefinitions.map(({ documentElement, elementContext }, i) => {
```
Comment (twschiller) [inline src/components/documentBuilder/DocumentList.tsx:70]: Somehow I'm getting an `Cannot read properties of undefined (reading 'map')` error here 

I think it happens when there's an error calculating the asyncState?

Comment (twschiller) [reply src/components/documentBuilder/DocumentList.tsx:88]: I updated this to display the error when the factory of useAsyncState fails

**Review by twschiller — APPROVED** [review]
@BALEHOK Looking good. I will create a separate issue for running mapArgs in a different context

**Pull request merged**